### PR TITLE
[MOCA-2821] Feat/batch staking

### PR DESCRIPTION
**Pull request by tonyfung99**

### CODE REVIEW

File: script/BatchDeposit.s.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    try {
+      const gasPrice = (await hre.ethers.provider.getFeeData()).gasPrice;
+
+      const estimatedGas = await simpleStaking.stakeBehalf.estimateGas(addresses, amounts);
+      console.log(
+        `Current gas price per unit: ${hre.ethers.formatUnits(gasPrice, "gwei")} txn gas unit: ${estimatedGas} total cost: ${hre.ethers.formatEther(
+          gasPrice * estimatedGas
+        )} token: ${hre.ethers.formatEther(await token.balanceOf(await signer.getAddress()))}`
+      );
+      // currently seems it's callable for owner only
```
Comment (jpopxfile) [inline script/BatchDeposit.s.js:73]: now callable by _updater

File: script/BatchDeposit.s.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+async function run(batchSize = 500, startAt = 0, endAt = 0) {
+  const deployerPrivateKey = process.env.PRIVATE_KEY;
+  const stakingAddress = process.env.STAKING_CONTRACT;
+  const tokenAddress = process.env.TOKEN_CONTRACT;
+
+  const signer = new hre.ethers.Wallet(deployerPrivateKey, hre.ethers.provider);
+
+  const simpleStaking = await hre.ethers.getContractAt("SimpleStaking", stakingAddress, signer);
```
Comment (jpopxfile) [inline script/BatchDeposit.s.js:34]: check not paused first, otherwise stakeBehalf will revert.

**Comment by jpopxfile** [comment]
After the script is done, is it possible to confirm that the correct amount has been staked on behalf for users into the contract?

File: script/BatchDeposit.s.js (hunk 1 above)
Comment (tonyfung99) [reply script/BatchDeposit.s.js:73]: ah yea forgot to update the comment, will do

**Pull request merged**